### ISSUE TITLE
Non-public profiles are not shown to auth users in demo instances

### ISSUE THREAD

**Issue by asaunier** [c2corg/v6_ui#793]
In dev instances (using pserve), non-public profiles are displayed as expected when authenticated.

In demo instances (using apache), even when browsing as an authenticated user, such pages keep returning "This profile is only available to authenticated users."

When requesting the UI ``profile/data`` service https://github.com/c2corg/v6_ui/blob/master/c2corg_ui/static/js/userprofile.js#L62-L65 the JWT token is correctly passed in an ``Authorization`` header but it seems no longer available when the UI backend receives the request. It is then not added to the API request which, as a result, rejects the request for non-public profiles.
https://github.com/c2corg/v6_ui/blob/master/c2corg_ui/views/profile.py#L90-L94

I have tried to temporarly deactivate the Basic Auth protection on the demo instance to make sure it was not related, but with no more success.

Short version: ``Authorization`` is not received by the UI user data service when served by Apache, OK when served by pserve.

**Comment by tsauerwein** [comment]
The API is also served with Apache and there we receive the `Authorization` header?

**Comment by asaunier** [comment]
Yes that's weird.
Investigating...

**Comment by asaunier** [comment]
I have found other people complaining that Apache had stripped the ``Authorization`` header out:
http://stackoverflow.com/questions/26549250/apache-strips-down-authorization-header
http://stackoverflow.com/questions/13387516/authorization-header-missing-in-django-rest-framework-is-apache-to-blame

It seems that Apache does so by default for security reasons. One may prevent this behaviour using
```
WSGIPassAuthorization On
```
as recommended at http://stackoverflow.com/a/18564966 or http://stackoverflow.com/a/26606276

and done in the API :)
https://github.com/c2corg/v6_api/blob/master/apache/wsgi.conf.in#L5

**Comment by asaunier** [comment]
PR #794 fixes the problem with Apache.

But there is then an ``Authorization`` header conflict when the Basic Auth is on: see https://github.com/c2corg/v6_ui/issues/771#issuecomment-256094288
